src/ripple/protocol/impl/STCurrency.cpp

```diff
@@ -0,0 +1,116 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2023 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/STCurrency.h>
+#include <ripple/protocol/jss.h>
+
+#include <ripple/basics/contract.h>
+
+namespace ripple {
+
+STCurrency::STCurrency(SField const& name) : STBase{name}
```

Comment:
Is this different from STIssue?

Reply:
`STIssue` contains a currency and an issuer. `STCurrency` contains just the currency.

Reply:
Why not just use a blob for it?

Reply:
The blob has to be a hexadecimal string. `STCurrency` can represent:
- A standard ISO-4217 currency, which is probably going to be used a lot for the `PriceUnit` and to the lesser extent used for the `Symbol` as well. The standard currency code is a three-letter string, which is user friendly.
- Any other asset represented as a 160-bit hexadecimal string.